zebra-state/src/service/finalized_state.rs

```diff
@@ -390,6 +396,19 @@ impl FinalizedState {
                 batch.zs_insert(history_tree_cf, height, history_tree);
             }
 
+            // Consensus rule: The block height of the genesis block is 0
+            // https://zips.z.cash/protocol/protocol.pdf#blockchain
+            if height == block::Height(0) {
+                batch.zs_insert(tip_chain_value_pool, height, ValueBalance::zero());
```

Comment:
This genesis code will never run, because the closure returns earlier if it gets a genesis block.

Reply:
Thanks, i removed that at https://github.com/ZcashFoundation/zebra/pull/2599/commits/7e50f1802508ff1da83940fa65b955d314165447